internal/api/approval/approval.go

```diff
@@ -170,7 +171,8 @@ func (a *Approval) ComputeApprovalStatus(ctx context.Context, pr *PR) (*Result,
 		for _, handle := range rule.ApprovingTeamHandles {
 			teamName, err := getTeamNameFromTeamHandle(teams, handle)
 			if err != nil {
```

Comment:
Wouldn't it be more obvious and explicit to test against certain error type prior assuming team handle is invalid?
```go
if err != nil {
  if errors.Is(ErrInvalidTeamHandle, err) {
    state.addInvalidTeamHandle(handle)
    continue
  }

  return err
}
```
Just to be 100% sure we don't dismiss any error we don't expect to be dismissed.

Reply:
@wojciecholszewski-form3 I am not sure if there is any other possible error to be handled here, if you check [getTeamNameFromTeamHandle](https://github.com/form3tech-oss/github-team-approver/blob/7e893f127c7dfa96124ad5035f9f4557d27f1eb2/internal/api/approval/approval.go#L353) you will see that it will throw an error if the condition is not satisfied and other than that just panic or fatal in the compilation itself. 

Not sure what is the logic behind of the exception but this would be fine either:

```go
if err != nil { 
    state.addInvalidTeamHandle(handle)
    continue
}
```
or 

```go
if err != nil { 
    state.addInvalidTeamHandle(handle)
    return nil, err
}
```

Or just create a new error type and return it [here](https://github.com/form3tech-oss/github-team-approver/blob/master/internal/api/approval/approval.go#L351)